### DISCUSSION TITLE
Making EEPROM selectable?

### DISCUSSION THREAD

**Discussion by ktelep** [TomNisbet/TommyPROM#26] [Ideas]
I lieu of having to define the EEPROM via #define, what are your thoughts on making the actual EEPROM selection more dynamic?   

For example I use this to program 28C65's (same as the C64, but with the READY line) and 28C256's, so I have to go back and forth reloading the code whenever I switch.   Not a huge deal, and I could just make a second programmer, but I don't believe it would be too difficult of functionality to add, but didn't want to write it if there was a specific reason it was being avoided.

My thoughts would be that you would probably define the chip family at build time, since more than likely that is defining your hardware build also, and then in the interface you could select either a canned device in the family (28C16, 28C64/65, 28C256) or individually set the capacity/blocksize/etc. on your own.

**Comment by TomNisbet** [comment]
There's no reason that wouldn't work.  I usually need to move wires when switching chip types so the recompile didn't seem like a big deal.  But I can see the case for using a few chips within the same family and just instantiating the prom object based on a menu selection or other input.  Let me know what you find or if you run into any problems.

I think I currently have 5 different versions built, either for completely different chip families or donewhile helping someone debug a problem.  Plus one with the Ben Eater design because so many people keep running into locked chips that I wrote a purpose-built unlocker for his hardware.

**Reply by TomNisbet** [reply]
It's also worth noting that code doesn't currently do anything with the device size parameter.  If you use the fill command on a 28C256 chip from 0000 to ffff it will just write the entire 32K twice.  If the parameters between the 28C256 and the 28C65 only differ by the device size then you can just use the same code without recompiling.  

If the two chips use different block write sizes then you could compile with the smaller block size and the only penalty would be that your larger chip would write slower because it would do more writes of smaller blocks than its optimal size.
